actors/miner/tests/util.rs

```diff
@@ -204,8 +261,1024 @@ impl ActorHarness {
+            Some(r) => r,
+            None => params.chain_commit_rand.clone(),
+        };
+        rt.expect_get_randomness_from_tickets(
+            DomainSeparationTag::PoStChainCommit,
+            params.chain_commit_epoch,
+            Vec::new(),
+            chain_commit_rand,
+        );
+        rt.expect_validate_caller_addr(self.caller_addrs());
+
+        let challenge_rand = Randomness(Vec::from([10, 11, 12, 13]));
+
+        // only sectors that are not skipped and not existing non-recovered faults will be verified
+        let mut all_ignored = BitField::new();
+        let mut all_recovered = BitField::new();
+        let dln = self.get_deadline(rt, deadline.index);
+        for p in &params.partitions {
+            let partition = dln.load_partition(&rt.store, p.index).unwrap();
+            let expected_faults = &partition.faults - &partition.recoveries;
+            let skipped = get_bitfield(&p.skipped);
+            all_ignored |= &(&expected_faults | &skipped);
+            all_recovered |= &(&partition.recoveries - &skipped);
+        }
+        let optimistic = all_recovered.is_empty();
+
+        // find the first non-faulty, non-skipped sector in poSt to replace all faulty sectors.
+        let mut good_info: Option<SectorOnChainInfo> = None;
+        for ci in &infos {
+            if !all_ignored.get(ci.sector_number) {
+                good_info = Some(ci.clone());
+                break;
+            }
+        }
+
+        // good_info == None indicates all the sectors have been skipped and PoSt verification should not occur
+        if !optimistic && good_info.is_some() {
+            let entropy = RawBytes::serialize(self.receiver.clone()).unwrap();
+            rt.expect_get_randomness_from_beacon(
+                DomainSeparationTag::WindowedPoStChallengeSeed,
+                deadline.challenge,
+                entropy.to_vec(),
+                challenge_rand.clone(),
+            );
+
+            let mut proof_infos = Vec::with_capacity(infos.len());
+            for ci in infos {
+                let mut si = ci.clone();
+                if all_ignored.get(ci.sector_number) {
+                    si = good_info.clone().unwrap();
+                }
+                let proof_info = SectorInfo {
+                    proof: si.seal_proof,
+                    sector_number: si.sector_number,
+                    sealed_cid: si.sealed_cid,
+                };
+                proof_infos.push(proof_info);
+            }
+
+            let vi = WindowPoStVerifyInfo {
+                randomness: challenge_rand.clone(),
+                proofs: params.proofs.clone(),
+                challenged_sectors: proof_infos,
+                prover: RECEIVER_ID,
+            };
+            let exit_code = match cfg.verification_exit {
+                Some(exit_code) => exit_code,
+                None => ExitCode::Ok,
+            };
+            rt.expect_verify_post(vi, exit_code);
```

Comment:
This is out of date with the go mock runtime in an analogous way to the bug discovered and fixed in this PR.  The mock runtime verify post expectation struct should contain a regular error instead of an exit code.

Reply:
it just checks to see if it is Ok, so in theory we could even change it to bool.

I am inclined to leave it as is for now, we can always revisit.

Reply:
Note that there is no such thing as a regular error in rust :)